src/sentry/utils/services.py

```diff
@@ -465,6 +470,64 @@ def setup(self) -> None:
             backend.setup()
 
 
+KeyFetch = Callable[[Context, str, Mapping[str, Any]], str | int]
+
+
+def make_writebehind_selector(
+    *, option_name: str, key_fetch: KeyFetch, move_to: str, move_from: str
+) -> Selector:
+    """
+    Generates a selector_func that will do write-behind delegation
+
+    The provided option_name is expected to have values between -1 and 1
+
+    -1.0 - 0.01 The move_from will be primary, while move_to will increasingly be added as a secondary.
+    At 0.0 - Only move_from will be used.
+    0.01 - 1.0 The move_to will increasingly be used as primary.
+
+    The `key_fetch` function gets the parameters expected by `Selector` and
+    is expected to return a consistent str|int that will be hashed for consistent
+    rollouts. If no consistent key exists you can use random number generation.
+
+    Example:
+
+    selector = make_writebehind_selector(
+        option_name="feature.rollout",
+        move_to="new",
+        move_from="old",
+        key_fetch=lambda *args: "a-consistent-key",
+    )
+    """
+
+    def selector(context: Context, method: str, callargs: Mapping[str, Any]) -> list[str]:
+        rollout_rate = options.get(option_name)
+        if rollout_rate == 0.0:
+            return [move_from]
+
+        key = key_fetch(context, method, callargs)
+        if isinstance(key, str):
+            intkey = int(hashlib.md5(key.encode("utf8")).hexdigest(), base=16)
+        else:
+            intkey = key
+
+        if not isinstance(intkey, int):
+            logger.error("make_writebehind_selector.invalid", extra={"received_type": type(intkey)})
+            return [move_from]
+
+        if rollout_rate < 0:
+            if (intkey % 10000) / 10000 < rollout_rate * -1.0:
+                return [move_from, move_to]
+            return [move_from]
+        else:
+            # rollout > 0
+            if (intkey % 10000) / 10000 < rollout_rate:
+                return [move_to, move_from]
+
+        return [move_from, move_to]
```

Comment:
It this accessible?

Reply:
It will be used when rollout > 0 but the current key is not in the rollout group. Looking at the tests again, there isn't coverage for this scenario. I'll expand tests.